pkgs/applications/science/misc/openrefine/update.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p nix nix-update
+
+set -euxo pipefail
+
+nix-update "${UPDATE_NIX_ATTR_PATH}"
+nix-update "${UPDATE_NIX_ATTR_PATH}.npmPkg" --version=skip
+
+FILE="$(nix-instantiate --eval -E 'with import ./. {}; (builtins.unsafeGetAttrPos "version" '"${UPDATE_NIX_ATTR_PATH}"').file' | tr -d '"')"
+
+MVNHASH_OLD=$(nix-instantiate --eval . -A "${UPDATE_NIX_ATTR_PATH}.mvnHash" | tr -d '"')
+MVNHASH_OLD_ESCAPED=$(echo "${MVNHASH_OLD}" | sed -re 's|[+]|\\&|g')
+FAKEHASH=$(nix-instantiate --eval . -A "lib.fakeHash" | tr -d '"')
+FAKEHASH_ESCAPED=$(echo "${FAKEHASH}" | sed -re 's|[+]|\\&|g')
+
+sed -E -i "s|${MVNHASH_OLD_ESCAPED}|${FAKEHASH}|g" "${FILE}"
+
+MVNHASH_NEW="$(nix-build . -A "${UPDATE_NIX_ATTR_PATH}" 2>&1 | tail -n10 | grep 'got:' | cut -d: -f2- | xargs echo || true)"
```

Comment:
Isn't there some better way to determine the new hash than trying to build with a fake hash and grepping the actual hash out of the error message?

Reply:
If there is, you'll have to show me it, and also show it to all the other authors of update scripts who use this technique ;)

I didn't invent this - look throughout nixpkgs for people grepping for `got:`

Reply:
Huh, indeed:
```
$ git ls-files '**/update.sh' | wc -l
208

$ git grep 'got:' | wc -l
54
```
So approximately one out of every four update scripts uses that. I guess the others (mostly findable by `git grep -L 'got:' '**/update.sh'`) only need hashes determinable by `nix-prefetch-url` or rely on `update-source-version` (which *also* [uses that grep-for-`got:` approach](https://github.com/NixOS/nixpkgs/blob/88e3ac79cb08011d02518df0bd29b0ad8ed5372c/pkgs/common-updater/scripts/update-source-version#L242), at least for some cases.)